imagebuildah/executor.go

```diff
@@ -522,7 +522,7 @@ func (b *Executor) buildStage(ctx context.Context, cleanupStages map[int]*StageE
 	// remove the intermediate/build containers, regardless of
 	// whether or not the stage's build fails.
 	// Skip cleanup if the stage has no instructions.
-	if b.forceRmIntermediateCtrs || !b.layers && len(stage.Node.Children) > 0 {
+	if b.forceRmIntermediateCtrs || !b.layers {
```

Comment:
The comment on the above line becomes invalid.  Does this regressing on #3025?

Reply:
Removed the comment, the test added in #3025 still exists on main, which is not regressed I have even tried the case mentioned in #3025 manually https://github.com/containers/buildah/blob/main/tests/bud.bats#L4299